example/custom_bot.py

```diff
@@ -50,63 +28,50 @@ def _preprocess(self):
                     bbox[3] *= DISPLAY_HEIGHT / SCREENSHOT_HEIGHT
                     bbox_bottom = [((bbox[0] + bbox[2]) / 2), bbox[3]]
                     unit['tile_xy'] = self._get_nearest_tile(*bbox_bottom)
-                    
-    def _click(self, x, y):
-        """
-        Clicks at the specified coordinates within the game window.
-        """
-        # Implement the actual click logic here (e.g., using pyautogui or another library)
-        pass  # Temporary placeholder
 
     def run(self):
-        """
-        Starts the bot and runs it in a loop.
-        """
-        print("Custom Bot is running...")
         while True:
-            # Check for end of game screen and pause
             if self.end_of_game_clicked:
                 if time.time() > self.pause_until:
+                    if not any(action.name == "Battle" for action in self.get_actions()):
+                        print(f"[{datetime.now().strftime('%H:%M:%S')}] [INFO] Cant find Battle button force game restart.")
+
+                        subprocess.run("adb shell am force-stop com.supercell.clashroyale", shell=True)
+                        time.sleep(1)
+                        subprocess.run("adb shell am start -n com.supercell.clashroyale/com.supercell.titan.GameApp", shell=True)
+
+                        print(f"[{datetime.now().strftime('%H:%M:%S')}] [INFO] Waiting 10 seconds...")
+                        time.sleep(10)
+
                     self.end_of_game_clicked = False
                 else:
-                    time.sleep(1.0)  # Wait until the pause is over
+                    time.sleep(1.0)
                     continue
 
-            # Update game state
             self.set_state()
 
-            # End of game detected, but no actions performed yet
             if self.state['screen'] == 'end_of_game':
-                self._click(*SCREEN_CONFIG['end_of_game']['click_coordinates'])
-                print("End of game detected, waiting for 10 seconds...")
-                self.pause_until = time.time() + 10  # Set pause for 10 seconds
-                self.end_of_game_clicked = True  # Mark the end of the game
-                time.sleep(2)  # Additional pause to ensure state is updated
-                continue  # Skip to the next loop iteration without performing actions
+                print(f"[{datetime.now().strftime('%H:%M:%S')}] [INFO] End of game detected. Waiting 10 seconds for battle button...")
```

Comment:
We can use a library like Loguru to do this logging easier. I'll update it in a future MR.

Reply:
Definitely open to suggestions for improvements, I'll look into Loguru after work tomorrow. I think this project will bring me some fun days in the future. :)